### ISSUE TITLE
make a simple privacy aware Camera app with HDR+ via the Pixel Visual Core

### ISSUE THREAD

**Issue by thestinger** [AndroidHardeningArchive/bugtracker#841]
We're going to need our own camera app using internal storage to store pictures and exposing them [with a document provider](https://developer.android.com/guide/topics/providers/document-provider.html). It won't bother with internal support for browsing the pictures which is where a fair bit of the complexity for the AOSP Camera2 app stems from. It can also make sure to avoid adding metadata that's more identifying than the pictures being taken from a given device variant.

The code for still shots can be based on https://github.com/google/pixelvisualcorecamera to take full advantage of modern camera capabilities including having HDR+ on the Pixel Visual Core for the Pixel 2 (XL) and future Pixel devices. It will need support for recording videos too.

It's unclear if we should start from AOSP Camera or from https://github.com/google/pixelvisualcorecamera. AOSP Camera is incredibly over-engineered. We'll definitely strip out support for anything not modern. It didn't even work properly on devices with camera3 API support before we made a tweak to it.

**Comment by thestinger** [comment]
Blocked by https://github.com/copperhead/bugtracker/issues/842 because it's not currently reliable enough and ends up stalling / crashing.

**Comment by thestinger** [comment]
There are substantial changes to storage in Android Q and I think it's best to wait until it's released before revisiting what needs to be done. I'm closing this for now rather than migrating it to the new issue tracker at https://github.com/GrapheneOS/os_issue_tracker/issues.
